src/metabase/driver/sql_jdbc.clj

```diff
@@ -117,19 +117,20 @@
   [_driver _semantic_type expr]
   (h2x/->timestamp expr))
 
-(defn- create-table-sql
-  [driver table-name col->type]
+(defn- create-table!-sql
```

Comment:
is it better to define `defmethod driver/create-table!` for `:redshift` rather than making a breaking change to `driver/create-table!` method?

now looking at this I also think maybe we should have `create-table-sql` as a defmulti.

Reply:
IMO the `driver/create-table!` method was designed incorrectly, including constraint information like primary keys in the `col->type` argument. One downside of that is its previous form it doesn't support multiple column primary keys. I'd prefer to fix that design here.

As for a `create-table-sql` multimethod, that's a good idea. It would separate the execution logic (which should be the same for all SQL drivers) from the SQL format, which might differ.

Reply:
I'd like to add the `create-table-sql` multimethod in another PR though, and batch it with more changes. If we do this for `create-table-sql` we should also do it for the other DDL multimethods like `driver/add-columns!` and `driver/drop-table!`, so the driver interface is consistent.

Reply:
yea bug what do you think about changing `col->type` to something like `col-definitions` tho?  like what we have in FieldDefinition for defdataset.

I'm a bit resistant about adding `primary-key` as a special case here; what if, in the future, we want to create a column to be indexed?

Reply:
+1 to that renaming, the meaning of that variable wasn't immediately clear to me

Reply:
> another idea: what do you think about changing col->type to something like col-definitions? like what we have in FieldDefinition for defdataset.

I like that idea. I believe `col->type` is left over from when we added primary keys, and we didn't change the name back then.

> I'm a bit resistant about adding primary-key as a special case here; what if, in the future, we want to create a column to be indexed?

Yes, I will change primary-key to be in a kwargs map

Reply:
Renamed col->type here 95c2d3a82295fe9ba2760e53ec10d4757ca9aa91
Changed primary-key to be in a kwargs map here b7d90136dd84ba2952df8b8d9cb9f06a2087c8ab

Reply:
sorry I wasn't clear before; I was thinking about structuring `column-definitions` like this.

```
[{:column-name "name", :column-type :text}
 {:column-name "id", :column-type: int, :primary-key? true}]
```

This way it'd be easier for us to add new property(e.g, indexed?, default-value ....) in the future.
It borrows the same idea that we use for `FieldDefinition` in tests: https://github.com/metabase/metabase/blob/master/test/metabase/test/data/interface.clj#L45.

wdyt about this approach?

Reply:
Ah right. I think that's a good idea but unnecessary as part of this change. The primary key should be separate from individual column items because in theory you should be able to create a primary key from an ordered list of columns

Reply:
okay, that makes sense